src/test/java/microsoft/exchange/webservices/data/RecurringAppointmentDateParsing.java

```diff
@@ -0,0 +1,42 @@
+package microsoft.exchange.webservices.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.junit.Test;
+
+public class RecurringAppointmentDateParsing extends BaseTest {
+
+
+	@Test
+	public void parseAPIRecurrenceDateFormat() throws ParseException {
+		String dateString = "2015-01-08Z";
+		Date newParsing = exchangeServiceBaseMock.convertStartDateToUnspecifiedDateTime(dateString);
+		Date oldParsing = convertStartDateToUnspecifiedDateTime(dateString);
+		assertEquals(oldParsing, newParsing);
+	}
+
+	@Test
+	public void parsePreviouslyUnbarsableRecurrenceDateFormat() {
+		String dateString = "2015-01-08-01:00";
+		Date parsed = exchangeServiceBaseMock.convertUniversalDateTimeStringToDate(dateString);
+		assertNotNull(parsed);

```

Comment:
maybe we should also assert the correct parsing here? Something like:
* year is 2015
* month / day are valid after parsing
* hour / minutes are correct too